### ISSUE TITLE
Firebase Auth with Serverpod serverResponse.userInfo! Null check operator used on a null value

### ISSUE THREAD

**Issue by tusharbhambere** [serverpod/serverpod#417]
I am Doing All Necessary Step From this documentation (https://docs.serverpod.dev/concepts/authentication#setting-up-sign-in-with-firebase)
All Checks
- Firebase connected Successfully
- Serverpod Server Running
- Database connected 
- Necessary Packages and CLI tool installed
- both firebase_service_account_key.json and google-services.json added thier right place

but at the end authentication successfully Done its throw error Null check operator used on a null value at  this line=>   serverResponse.userInfo!,
 // Store the user info in the session manager.
                    var sessionManager = await SessionManager.instance;
                    await sessionManager.registerSignedInUser(
                      serverResponse.userInfo!,
                      serverResponse.keyId!,
                      serverResponse.key!,
                    );
And also User is successfully Created on firebase i can seen from firebase console

**Comment by tusharbhambere** [comment]
![image](https://user-images.githubusercontent.com/77154182/204269188-52ba99f8-7c6b-45e2-8d31-bab68789c414.png)

Trying to Create Account

**Comment by yahu1031** [comment]
@tusharbhambere How are you signing in as the user? and how are you collecting `serverResponse` ? Could you please share the code block for it?

**Comment by tusharbhambere** [comment]
for connecting firebase i added serverpod_auth_firebase_flutter which provider me this SignInWithFirebaseButton and in caller i simply pass caller

  SignInWithFirebaseButton(
                caller: ref.watch(sessionManegerProvider).caller,
                onFailure: () {
                  log('fail');
                },
                onSignedIn: () {
                  log('success');
                },
                providerConfigs: const [
                  EmailProviderConfiguration(),
                  // PhoneProviderConfiguration()
                  // GoogleProviderConfiguration(
                  //     clientId:
                  //         '')
                ]),

**Comment by yahu1031** [comment]
Have you gone through [session manager setup](https://docs.serverpod.dev/concepts/authentication#setting-up-sign-in-with-firebase:~:text=Next%2C%20you%20need%20to%20set%20up%20a%20SessionManager%2C%20which%20keeps%20track%20of%20the%20user%27s%20state)? Just asking...

**Comment by tusharbhambere** [comment]
Not Setup above email and google part
just follow Firebase part (https://docs.serverpod.dev/concepts/authentication#setting-up-sign-in-with-firebase)
Yes read many time but for this part
 **Server-side, you need to create a set of authentication keys in the [Firebase console](https://console.firebase.google.com/), download the JSON file, rename it to firebase_service_account_key.json and place it in the config folder.**

i was go to firebase console>project> service account > generate new key and simply change that file name and paste in server>config folder as firebase_service_account_key.json

**Comment by yahu1031** [comment]
Okie, I'm trying to reproduce it.

**Comment by II11II** [comment]
I could reproduce it on v0.9.19

**Comment by vlidholt** [comment]
I will go through the complete authentication process before the 1.0 release to ensure everything is working!

**Comment by avatarnguyen** [comment]
Hey all,

anyone already have an idea what could caused this issue?

For me I am able to login with Firebase and get FirebaseUser but the `serverResponse.success` is `false` and `userInfo` is always null.

```
AuthStateChangeAction<SignedIn>((context, state) async {
...

  final user = state.user!;
         
  final idToken = await user.getIdToken();
            
  final serverResponse = await caller.firebase.authenticate(idToken);

})
```

**Comment by vlidholt** [comment]
@avatarnguyen Have you added the database tables for the auth module?

**Comment by avatarnguyen** [comment]
@vlidholt I think I did. but not sure whether I miss something. here is all the table I have:

<img width="301" alt="Screenshot 2023-04-30 at 11 40 41" src="https://user-images.githubusercontent.com/39520774/235346439-c6c42086-b95e-4e8d-8fe7-32deceb709d0.png">

**Comment by inf0rmatix** [comment]
Adding the real message from firebase admin sdk in the server error response would be the best option imho. I had the same problem and only found out that somehow my firebase service account json was corrupt by manually modifying the serverpod code to print the error from firebase admin sdk.

**Comment by Isakdl** [comment]
I have taken a look at this, so I think most likely the issue is that the json key is corrupt, I would try to create a new one and delete the old as the first step.

Second, we had an issue where we would only allow providers that provided an email with the user, I.E. phone authentication didn't work. I fixed this in the PR above.

I have also added better logging so in case the json key is the issue you will now get an error logged on the server.
